Bottom: 7b7885454a30f43c9694aa683b2b900201dafca1
Top:    b5938f8796f372c81fa9bedfc4ff5b69132f362e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 23:56:11 +0000

Debug wonky copper polygon generation


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 2cdb502..ab316e7 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1187,19 +1187,19 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
         {
           fprintf (stderr, "Accumulating elements from layer %i\n", GetLayerNumber (PCB->Data, layer));
 
-          r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
-          r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
-          r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
+//          r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
+//          r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
+//          r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
           r_search (layer->polygon_tree, &bounds, NULL, polygon_copper_callback, &info);
         }
       END_LOOP;
 
-      fprintf (stderr, "Accumulating pin + via pads\n");
-      r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
-      r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
+//      fprintf (stderr, "Accumulating pin + via pads\n");
+//      r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
+//      r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
 #endif
 
-#if 1
+#if 0
       if (group == top_group ||
           group == bottom_group)
         {
@@ -1252,7 +1252,7 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
                                 copper_appearance,
                                 NULL));
 
-
+      break;
     }
 
 
@@ -1263,7 +1263,11 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
 //  PCB->Data->outline_valid = true;
 //  gui->invalidate_all ();
 
-  poly_Free (&info.poly);
+  /* ASSUME THERE IS A POLYGON WHERE WE KNOW WE PUT ONE... */
+  ((PolygonType *)PCB->Data->Layer[1].Polygon->data)->Clipped = info.poly;
+  gui->invalidate_all ();
+
+//  poly_Free (&info.poly);
 
   return objects;
 }
diff --git a/src/polygon1.c b/src/polygon1.c
index 5c09126..2372153 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -39,6 +39,7 @@
       are marked
 */
 
+#undef NDEBUG
 #include	<assert.h>
 #include	<stdlib.h>
 #include	<stdio.h>
@@ -109,13 +110,13 @@ int vect_inters2 (Vector A, Vector B, Vector C, Vector D, Vector S1,
 
 #define error(code)  longjmp(*(e), code)
 
-#undef DEBUG_INTERSECT
-#undef DEBUG_LABEL
-#undef DEBUG_ALL_LABELS
-#undef DEBUG_JUMP
-#undef DEBUG_GATHER
-#undef DEBUG_ANGLE
-#undef DEBUG
+#define DEBUG_INTERSECT
+#define DEBUG_LABEL
+#define DEBUG_ALL_LABELS
+#define DEBUG_JUMP
+#define DEBUG_GATHER
+#define DEBUG_ANGLE
+#define DEBUG
 #ifdef DEBUG
 #define DEBUGP(...) pcb_fprintf(stderr, ## __VA_ARGS__)
 #else
